src/scales/scale.time.js

```diff
@@ -506,15 +529,22 @@ module.exports = function(Chart) {
 		 * Function to format an individual tick mark
 		 * @private
 		 */
-		tickFormatFunction: function(tick, index, ticks) {
+		tickFormatFunction: function(tick, index, ticks, displayFormatOverride) {
 			var me = this;
 			var options = me.options;
+			var ticksOpts = options.ticks;
 			var time = tick.valueOf();
 			var majorUnit = me.majorUnit;
-			var majorFormat = me.majorDisplayFormat;
-			var majorTime = tick.clone().startOf(me.majorUnit).valueOf();
+			var displayFormat = options.time.displayFormats[me.unit];
+			var majorFormat = ticksOpts.major.displayFormats[majorUnit];
+			var minorFormat = ticksOpts.minor.displayFormats[me.unit];
+			var majorTime = tick.clone().startOf(majorUnit).valueOf();
 			var major = majorUnit && majorFormat && time === majorTime;
-			var formattedTick = tick.format(major ? majorFormat : me.displayFormat);
+			var formattedTick = tick.format(
+				displayFormatOverride !== undefined ? displayFormatOverride :
+				options.time.unit ? displayFormat :
+				major ? majorFormat :
+				minorFormat);
```

Comment:
It's not correct because we should always use `minor` if it's not a major tick (`minor` options inherit from the root ones). It's still a breaking change because people who didn't explicitly specify a `unit` will see their graph considerably changed. For example, if the user defined formats that works great in their project, this will be overridden by the new defaults minor:
```javascript
{
   // no explicit unit
   defaultFormats: {
       //  overridden by minor [breaking change]
    }
}
```

But also this case doesn't work if the user specify both `unit` and `defaultFormats`:
```javascript
{
    unit: 'days',
    minor: {
         defaultFormats: {
             // ignored [bug]
         }
    }
}
```